### PR TITLE
Initial Dragonflight compatibility

### PR DESCRIPTION
**Pull request by Silarn**

### CODE REVIEW

**Comment by VixusFoxy** [comment]
Confirming that this set of changes works.

**Comment by Silarn** [comment]
I'm going to do a cleanup pass, I missed some semicolons and should change the version string back to the replacement string...

File: Core/Preview.lua [hunk 1]
```diff
@@ -57,8 +57,12 @@ end
...
 local function resizeOnMouseDown(self)
 	local f = self:GetParent();
-	f:SetMinResize(335,385);
-	f:SetMaxResize(GetScreenWidth(),GetScreenHeight());
+	if f.SetResizeBounds then
+		f:SetResizeBounds(335, 385, GetScreenWidth(), GetScreenHeight());
+	else
+		f:SetMinResize(335,385);
+		f:SetMaxResize(GetScreenWidth(), GetScreenHeight());
+	end
```
Comment (Lombra) [inline Core/Preview.lua:65]: I'm guessing the `if` is for backwards compatibility? I don't think there's much sense in having that here, as no such effort has been done for the the rest of the codebase.

File: Core/Options.lua [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿local MogIt,mog = ...;
+local MogIt,mog = ...;
```
Comment (Lombra) [inline Core/Options.lua:1]: Can't see what's changed here. Whitespace?

File: Core/Preview.lua (hunk 1 above)
Comment (Silarn) [reply Core/Preview.lua:65]: In theory yes. I wasn't sure if this was something you wanted or not, if so I can look into adding more compatibility checks.

File: Core/Options.lua (hunk 2 above)
Comment (Silarn) [reply Core/Options.lua:1]: Must be whitespace, maybe the line ending? My IDE does tend to clean up whitespace.

File: Core/Preview.lua (hunk 1 above)
Comment (Lombra) [reply Core/Preview.lua:65]: To be clear, I didn't mean that *you* had made no such effort! But I would rather leave this out, for now.

**Comment by Lombra** [comment]
Appreciated! Not really fully used to dealing with pull requests, yet.

**Comment by Silarn** [comment]
All good. Let me know how you want to deal with the backwards/cross compatibility stuff. I guess this doesn't really exist in Classic so it's probably not necessary. Some things like SetTitle seem to already exist in the older interface code, others definitely don't.

There are some deprecated function translations for some of the stuff that was changed so I don't necessarily have to convert those to the newer method just yet. I imagine those may go away at some point.

**Comment by Lombra** [comment]
Let's skip the compatibility logic for now. Maybe if they decide to make Cataclysm classic it could be worth considering.

**Comment by Silarn** [comment]
Should be able to get to this in an hour or two.

**Comment by Starvoicereason** [comment]
How do I USE these new fun things that have apparently been made?? I need MogIt for my sanity.

**Comment by VixusFoxy** [comment]
> How do I USE these new fun things that have apparently been made?? I need MogIt for my sanity.

**If you don't know how, you're best off waiting** until the addon is updated through Curseforge.
That, or you can go be brave and experiment by pulling a fork that has these changes in it and overwriting your local files with them.

File: Core/Core.lua [hunk 3]
```diff
@@ -349,36 +349,39 @@ function mog:ADDON_LOADED(addon)
...
 						mog:AddToPreview(select(6, C_TransmogCollection.GetAppearanceSourceInfo(source)), preview)
 					end
 					return
 				end
-				orig_OnMouseUp(self, button)
-			end)
-		end
+				orig_OnClick(self, button2);
+			end);
+		end, self, false);
```
Comment (Lombra) [inline Core/Core.lua:380]: I reckon we might as well pass `true` here in order to catch any previously created frames, for good measure.

**Review by Lombra — CHANGES_REQUESTED** [review]

File: Core/Core.lua (hunk 3 above)
Comment (Silarn) [reply Core/Core.lua:380]: Based on my testing, nothing is actually initialized when we set up this event handler. My concern was that we'd potentially just end up rehooking all of the frames every time new ones get initialized.

Comment (Silarn) [reply Core/Core.lua:380]: I guess we could attach a property to indicate if the frame was already initialized by mogit.

Comment (Lombra) [reply Core/Core.lua]: That would happen regardless of the value this parameter if I understand correctly. But now that you mention it, the `Acquired` callback might be more suitable, unless script handlers are reset on `Initialize`.

**Pull request merged**